extension-client/src/api/index.ts

```diff
@@ -83,7 +83,8 @@ export async function pollState(): Promise<string> {
             getWorkspace(wsData.id, credentials)
                 .then((res: any) => {
                     const status = res.status;
-                    if (continueStates.includes(status)) {
+                    const locked = res.workspace_status.locked;
+                    if (continueStates.includes(status) && !locked) {
```

Comment:
boolean variables follow [naming convention](https://dev.to/michi/tips-on-naming-boolean-variables-cleaner-code-35ig), so please use descriptive variable names such as - `isWorkspaceLocked`